addons/website/static/src/snippets/s_popup/000.js

```diff
@@ -79,6 +79,10 @@ const PopupWidget = publicWidget.Widget.extend({
      * @override
      */
     start: function () {
+        const hash = window.location.hash;
+        if (hash && this.el.querySelector(".modal").id === hash.substring(1)) {
```

Comment:
Should not you check the data-display to be sure / be more explicit here? Actually, I think there is better to do here: right now, there is an inconsistency: the "autoshow popup if the hash is in the URL" is handled by the popup code but the "show popup if the hash is in the URL on button click" is handled by the animate click stuff. IMO, both could be here in the popup code, that would actually be more generic (not only work with `<a>` elements) but the opening of the popup would also go though `_showPopup` here which could be overridden by some custom code. So let's do that: you can listen to the "hashchange" event here to detect when a button with an anchor is clicked. Arguably we should also do that instead of the anchorSlide code too, but that can wait.

Reply:
> Should not you check the data-display to be sure / be more explicit here?

ok :+1: , I also added some comments to explain the code